doc/programming_guide/gui/concept.rst

```diff
@@ -127,99 +159,317 @@ Size hint support
+
+Text area
+^^^^^^^^^
+
+**Name**: :py:class:`~arcade.gui.UITextArea`
+
+A text area is a scrollable text widget. A user can scroll the mouse to view a
+rendered text document. **This does not support editing text**. Think of it as
+a scrollable label instead of a text field.
+
+``width`` and ``height`` allocate a size for the text area. If text does not
+fit within these dimensions then only part of it will be displayed. Scrolling
+the mouse will display other sections of the text incrementally. Other
+parameters include ``multiline`` and ``scroll_speed``. See
+:py:attr:`~pyglet.text.layout.ScrollableTextLayout.view_y` on scroll speed.
+
+Use ``layout`` and ``doc`` to get the pyglet layout and document for the
+text area, respectively.
 
 .. _UIEvent:
 
-UIEvents
-========
-
-UIEvents are fully typed dataclasses, which provide information about a event effecting the UI.
-
-All pyglet window events are converted by the UIManager into UIEvents and passed via dispatch_event
-to the ``on_event`` callbacks.
+User-interface events
+=====================
+
+Arcade's GUI events are fully typed dataclasses, which provide information
+about an event affecting the UI.
+
+All pyglet window events are converted by the
+:py:class:`~arcade.gui.UIManager` into :class:`UIEvents` and passed via
+:py:meth:`~pyglet.event.EventDispatcher.dispatch_event` to the
+:py:meth:`~arcade.gui.UIWidget.on_event` callbacks.
+
+Widget-specific events (such as :py:class:`~arcade.gui.UIOnClickEvent` are
+dispatched via ``on_event`` and are then  dispatched as specific event types
+(like ``on_click``).
+
+A full list of event attributes is shown below.
+
++---------------------------+-----------------------------------------+
+|Event                      |Attributes                               |
++===========================+=========================================+
+|``UIEvent``                |None                                     |
++---------------------------+-----------------------------------------+
+|``UIMouseEvent``           |``x``, ``y``                             |
++---------------------------+-----------------------------------------+
+|``UIMouseMovementEvent``   |``dx``, ``dy``                           |
++---------------------------+-----------------------------------------+
+|``UIMousePressEvent``      |``dx``, ``dy``, ``button``, ``modifiers``|
++---------------------------+-----------------------------------------+
+|``UIMouseDragEvent``       |``dx``, ``dy``                           |
++---------------------------+-----------------------------------------+
+|``UIMouseScrollEvent``     |``scroll_x``, ``scroll_y``               |
++---------------------------+-----------------------------------------+
+|``UIKeyEvent``             |``symbol``, ``modifiers``                |
++---------------------------+-----------------------------------------+
+|``UIKeyReleaseEvent``      |None                                     |
```

Comment:
Should we mention, that UIKeyReleaseEvent inherits from UIKeyEvent and also has `symbol` and `modifiers`?